src/waypoint.jsx

```diff
@@ -333,13 +352,30 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.noWrapper) {
+      const child = React.Children.only(this.props.children);
+      const ref = (node) => {
+        this.refElement(node);
+        if (this.props.children.ref) {
+          this.props.children.ref(node);
+        }
+      };
+      return React.cloneElement(child, { ref });
+    }
+
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
+    }
+
     // We need an element that we can locate in the DOM to determine where it is
     // rendered relative to the top of its context.
     return <span ref={this.refElement} style={{ fontSize: 0 }} />;
   }
 }
 
 Waypoint.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
+  noWrapper: PropTypes.bool,
```

Comment:
Yeah, I struggled with that. Ultimately figured it would be confusing for too many users, especially beginners, since they would have to pass a single React Class-based Component or DOM Element. 

The "INVALID" examples I included in the docs are all perfectly reasonable things to do, and allowing them makes ReactWaypoint more approachable. 

Thoughts?